WIP POST test is failing

scmountain opened the pull request:
WIP - 

We're not sure why we can't get the post request to update app.locals.foods. We have been following Micah's lesson to build out the first section of our node application. We've checked the syntax, looked at the Express documentation, and console.logged out the message, app.locals.storage, and a completed post request. 

We inserted a comment into the code that points out the exact place. 

We're at a loss, please let us know if we're missing something or how to move forward. Thanks so much!

scmountain committed:
WIP POST test is failing

scmountain commented on test/server-test.js:
```diff
@@ -26,16 +28,61 @@ describe('Server', function(){
+      });
+    });
+
+    it('should recieve and store data', function(done){
```
@case-eee @neight-allen we cannot get this test to pass. When we console log in, we see that our message is not being saved. it is still an empty object. ``` app.locals.foods is {} ```

case-eee replied:
@scmountain it looks like you've defined two routes as a post to '/api/foods' - one on line 28 and one on line 32 in your `server.js` file. could that be part of the problem?

are you sure the code that you expect to be executing when you run this test is being executed?

maxglassie committed:
Deletes extraneous post request in server.js

We're currently not understanding why the test is passing, even when it shouldn't.
The plan is to move on to implementing an SQL database and not using app.locals
for storage.

maxglassie committed:
Knex

Activates knex, builds out the database file, sets up database configuration
for database-spike, and modifies package.json, there was a mistake with the
test environment.

maxglassie committed:
Builds out the database-spike file

maxglassie committed:
Completes functionality for GET a food item

Builds out the test for returning an item from the foods table. Test passes, next
step is to create a model and controller structure. Needed to add the configuration at the top of
server.js to get the database setup properly. Two skipped POST request tests.

maxglassie committed:
Refactors GET request to MVC format

Refactors the GET request in server.js and creates a model and controller.
There is some tricky requiring of files / variables that occurs flowing from the model to
the controller and to the server file.
